pkg/rekor/identity.go

```diff
@@ -100,8 +104,77 @@ func (e *IdentityEntry) String() string {
 	return strings.Join(parts, " ")
 }
 
+// parseObjectIdentifier parses a string representing an ObjectIdentifier in dot notation
+// and converts it into an asn1.ObjectIdentiifer.
+func parseObjectIdentifier(oid string) (asn1.ObjectIdentifier, error) {
+	if len(oid) == 0 {
+		return nil, errors.New("could not parse object identifier: empty input")
+	}
+	nodes := strings.Split(oid, ".")
+	objectIdentifier := make([]int, len(nodes))
+	for i, node := range nodes {
+		if strings.TrimSpace(node) == "" {
+			return nil, errors.New("could not parse object identifier: no characters between two dots")
+		}
+		intNode, err := strconv.Atoi(node)
+		if err != nil {
+			return nil, err
+		}
+		objectIdentifier[i] = intNode
+	}
+	return asn1.ObjectIdentifier(objectIdentifier), nil
+}
+
+// mergeOIDMatchers groups all OID matchers from OIDMatchers, FulcioExtensions, and CustomOIDs into one slice of OIDMatchers
+func mergeOIDMatchers(mvs MonitoredValues) ([]identity.OIDMatcher, error) {
+	fulcioOIDMatchers, err := mvs.FulcioExtensions.RenderFulcioOIDMatchers()
+	if err != nil {
+		return nil, fmt.Errorf("error rendering OID Matchers from Fulcio OID extensions: %w", err)
+	}
+	// map of all OID extensions to all associated matching extension values
+	oidMap := make(map[string]map[string]bool)
+	// dedup OID extensions and associated values through one mapping
+	for _, oidMatcher := range mvs.OIDMatchers {
+		oidMatcherString := oidMatcher.ObjectIdentifier.String()
+		oidMap[oidMatcherString] = make(map[string]bool)
+		for _, extValue := range oidMatcher.ExtensionValues {
+			oidMap[oidMatcherString][extValue] = true
+		}
+	}
+	for _, oidMatcher := range fulcioOIDMatchers {
+		oidMatcherString := oidMatcher.ObjectIdentifier.String()
+		oidMap[oidMatcherString] = make(map[string]bool)
+		for _, extValue := range oidMatcher.ExtensionValues {
+			oidMap[oidMatcherString][extValue] = true
+		}
+	}
+	// convert map into list of OIDMatchers
+	var allMatchers []identity.OIDMatcher
+	for oidExtension, extValueMap := range oidMap {
+		parsedOID, err := parseObjectIdentifier(oidExtension)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing OID from extension: %w", err)
+		}
+		var extValues []string
+		for extValue := range extValueMap {
+			extValues = append(extValues, extValue)
+		}
+		allMatchers = append(allMatchers, identity.OIDMatcher{
+			ObjectIdentifier: parsedOID,
+			ExtensionValues:  extValues,
+		})
+	}
+
+	return allMatchers, nil
+}
+
 // MatchedIndices returns a list of log indices that contain the requested identities.
 func MatchedIndices(logEntries []models.LogEntry, mvs MonitoredValues) ([]IdentityEntry, error) {
+	allOIDMatchers, err := mergeOIDMatchers(mvs)
+	if err != nil {
+		return nil, err
+	}
+	mvs.OIDMatchers = allOIDMatchers
```

Comment:
While this works fine for this use case, `mvs` is a copy of a struct, so you aren't updating the `MonitoredValues` struct passed in when setting `mvs.OIDMatchers`. If you expect that `MatchedIndices` should update `OIDMatchers` for other APIs as well, then you'd want to update `mvs` to pass by pointer.

Reply:
hmm it does seem that if we have `MatchedIndices` update `OIDMatchers` within `mvs` that feels like a bit of a side effect/unclear behavior? WDYT about keeping this for now and adding a `#TODO` for the incoming `MatchedIndices`/`RunConsistencyCheck` refactor to add a method to preprocess `OIDMatchers` before passing that into `MatchedIndices`?

Reply:
Good with me! I didn't know if you wanted to have the side-effect of this method mutating OIDMatchers - sounds like no, so this looks good.